src/fire-event.ts

```diff
@@ -143,3 +146,14 @@ fireEvent.scroll = (element: ReactTestInstance, ...data: unknown[]) =>
   fireEvent(element, 'scroll', ...data);
 
 export default fireEvent;
+
+function setNativeStateIfNeeded(element: ReactTestInstance, eventName: string, value: unknown) {
```

Comment:
Should it be done for fireEvent at all? I think it's a very good idea for userEvent but I think fireEvent should remain as minimal as possible, even though there is the shorthand for `fireEvent.changeText`, making it do more than calling the `onChangeText` nurtures the confusion around the differences between fireEvent and userEvent imo. And here it would not work with fireEvent(element, 'change') which also adds confusion

Reply:
I would keep it, as many existing codebases rely on Fire Event, so restricting this to User Event, would prevent them from accessing the benefits of native state simulation.

I would guess that 90% of Fire Event users use `fireEvent.changeText` to change text instead of building custom event object for `fireEvent(element, "change", { event... })`. If someone complains that we do not support it, then we can add it here.

Reply:
Existing codebases indeed rely on `fireEvent` but they do not use this feature so it would not break anything for them. You're right that it would mean they don't have access to that feature but that's only true if they don't start using `userEvent`. I think we should see `fireEvent` as a deprecated API, not really deprecated but only to be used for specific use cases that `userEvent` does not cover and as such I would not add new features to it, otherwise I don't know if it would be clear for users which API they should use and why they should migrate to `userEvent`

Reply:
Both Fire Event and User Event are supported APIs, just like in RTL. While User Event provides more realistic testing by emitting multiple events, it's still fine to stay on Fire Event, e.g. when having a large test base using it. There is no cost for us to keep supporting Fire Event and add small enhancements like this. Shipping simulated native state for only User Event and Jest matchers but not Fire Event would cause confusion.